packages/hardhat-core/src/types/config.ts

```diff
@@ -136,15 +134,15 @@ export type HardhatNetworkChainsConfig = Map<
 export interface HardhatNetworkConfig {
   chainId: number;
   from?: string;
-  gas: "auto" | number;
-  gasPrice: "auto" | number;
+  gas: "auto" | bigint;
+  gasPrice: "auto" | bigint;
   gasMultiplier: number;
-  initialBaseFeePerGas?: number;
+  initialBaseFeePerGas?: bigint;
   hardfork: string;
   mining: HardhatNetworkMiningConfig;
   accounts: HardhatNetworkAccountsConfig;
-  blockGasLimit: number;
-  minGasPrice: BN;
+  blockGasLimit: bigint;
+  minGasPrice: bigint;
```

Comment:
I think this can be an exception, as we really want to remove `BN`, and this field is not super common